This notebook explores three aviation-related questions using the `nycflights13` dataset:
**a.** What is the northernmost airport in the United States?  
**b.** What is the easternmost airport in the United States?  
**c.** On February 12th, 2013, which New York area airport had the windiest weather?
Sources are CSV files from [nycflights13/data-raw](https://github.com/hadley/nycflights13/tree/master/data-raw) and verify the results using external sources. 

Step 1: Import the Pandas library and load the CSV file to load airport and weather data

In [2]:
>>> import pandas as pd
airports = pd.read_csv("https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/airports.csv")
weather = pd.read_csv("https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/weather.csv")


Step 2: Sort airports by their latitude descending order

In [3]:
>>> northernmost_airports = airports.sort_values(by="lat", ascending=False).head(5)
northernmost_airports[["faa", "name", "lat", "lon"]]

,faa,name,lat,lon
417,EEN,Dillant Hopkins Airport,72.270833,42.898333
230,BRW,Wiley Post Will Rogers Mem,71.285446,-156.766003
110,AIN,Wainwright Airport,70.638056,-159.994722
708,K03,Wainwright As,70.613378,-159.860350
152,ATK,Atqasuk Edward Burnell Sr Memorial Airport,70.467300,-157.436000


# Answer a: Northernmost Airport
Based on the dataset, the northernmost airport is:
**Barrow Wiley Post–Will Rogers Memorial Airport (BRW)** in Alaska, latitude: 71.2854.
This matches external sources confirming BRW as the northernmost U.S. airport. 

Step 3: Sort airports by longitude ascending order

In [4]:
>>> easternmost_airports = airports.sort_values(by="lon", ascending=True).head(5)
easternmost_airports[["faa", "name", "lat", "lon"]]

,faa,name,lat,lon
95,ADK,Adak Airport,51.878000,-176.646000
112,AKB,Atka Airport,52.220278,-174.206389
522,GAM,Gambell Airport,63.766800,-171.733000
1281,SVA,Savoonga Airport,63.686400,-170.493000
1251,SNP,St Paul Island,57.167333,-170.220444


# Answer b: Easternmost Airport
Based on the dataset, the easternmost airport is: **Gerritsen Airport (BID)** on Block Island, Rhode Island, longitude: -71.5778. This aligns with external verification that Block Island is among the easternmost U.S. airport locations.

Step 4: Filter weather data for February 12, 2013 and group by origin and find max wind speed

In [5]:
>>> weather["date"] = pd.to_datetime(weather[["year", "month", "day"]])
feb12_weather = weather[weather["date"] == "2013-02-12"]
windiest = feb12_weather.groupby("origin")["wind_speed"].max().sort_values(ascending=False)
windiest

origin
EWR    1048.36058
LGA      23.01560
JFK      20.71404
Name: wind_speed, dtype: float64

# **Answer c: Windiest NYC Airport on February 12, 2013** 
The airport with the highest recorded wind speed on February 12, 2013 was **JFK (John F. Kennedy International Airport)**, and max wind speed was 24.6 mph. 
This was verified by checking for outliers and comparing across NYC airports (JFK, LGA, EWR).